pandas/core/arrays/arrow/array.py

```diff
@@ -1980,48 +2016,124 @@ def _str_removesuffix(self, suffix: str):
+                    for chunk in self._pa_array.iterchunks()
+                ]
+            )
         )
 
-    def _str_encode(self, encoding, errors: str = "strict"):
-        raise NotImplementedError(
-            "str.encode not supported with pd.ArrowDtype(pa.string())."
+    def _str_encode(self, encoding: str, errors: str = "strict"):
+        return type(self)(
+            pa.chunked_array(
+                [
+                    [
+                        None
+                        if val.as_py() is None
+                        else val.as_py().encode(encoding, errors)
+                        for val in chunk
+                    ]
+                    for chunk in self._pa_array.iterchunks()
+                ]
+            )
         )
 
     def _str_extract(self, pat: str, flags: int = 0, expand: bool = True):
         raise NotImplementedError(
             "str.extract not supported with pd.ArrowDtype(pa.string())."
         )
 
-    def _str_findall(self, pat, flags: int = 0):
-        raise NotImplementedError(
-            "str.findall not supported with pd.ArrowDtype(pa.string())."
+    def _str_findall(self, pat: str, flags: int = 0):
+        regex = re.compile(pat, flags=flags)
+        return type(self)(
+            pa.chunked_array(
+                [
+                    [
+                        None if val.as_py() is None else regex.findall(val.as_py())
+                        for val in chunk
+                    ]
+                    for chunk in self._pa_array.iterchunks()
+                ]
+            )
         )
 
     def _str_get_dummies(self, sep: str = "|"):
-        raise NotImplementedError(
-            "str.get_dummies not supported with pd.ArrowDtype(pa.string())."
-        )
+        split = pc.split_pattern(self._pa_array, sep).combine_chunks()
+        uniques = split.flatten().unique()
+        uniques_sorted = uniques.take(pa.compute.array_sort_indices(uniques))
+        result_data = []
+        for lst in split.to_pylist():
+            if lst is None:
+                result_data.append([False] * len(uniques_sorted))
+            else:
+                res = pc.is_in(uniques_sorted, pa.array(set(lst)))
+                result_data.append(res.to_pylist())
+        result = type(self)(pa.array(result_data))
+        return result, uniques_sorted.to_pylist()
 
-    def _str_index(self, sub, start: int = 0, end=None):
-        raise NotImplementedError(
-            "str.index not supported with pd.ArrowDtype(pa.string())."
+    def _str_index(self, sub: str, start: int = 0, end: int | None = None):
+        return type(self)(
+            pa.chunked_array(
+                [
+                    [
+                        None
+                        if val.as_py() is None
+                        else val.as_py().index(sub, start, end)
+                        for val in chunk
+                    ]
+                    for chunk in self._pa_array.iterchunks()
+                ]
```

Comment:
is it worth making a helper for this pattern so this can just be

```
def _str_index(...)
    predicate = lambda x: x.index(sub, start, end)
    return self._helper_whatever(predicate)
```